### DISCUSSION TITLE
Question about actual simulation time step

### DISCUSSION THREAD

**Discussion by cooluser26** [epochpic/epoch#591] [Q&A]
Hello,

In the simulation, we can specify the time interval between output dumps but what is the actual simulation time step? (I think it's determined automatically?). And how does the actual simulation time step compare to the electron plasma frequency? I'm asking since I am just going through the process of what I should consider to ensure the simulation converges sufficiently. 

Thank you ( and sorry for so many questions!),

William

**Comment by Status-Mirror** [comment]
Hey William,

The default EPOCH time-step changes with simulation options and parameters, and is set in the subroutine `set_dt` in `src/housekeeping/setup.F90`. In a 2D-PIC code, the default time-step is `dx * dy / SQRT(dx**2 + dy**2) / c`, where `dx` and `dy` are the cell-sizes in `x` and `y`. This time-step ensures particles cannot pass multiple cells in a single PIC-cycle, and is unrelated to the electron plasma frequency. Changing things like the field order or field solver will modify this default.

In practice, you cannot make the time-step longer or the PIC algorithm will fail. You can make the timestep shorter by using the `dt_multiplier` key in the [control block](https://epochpic.github.io/documentation/input_deck/input_deck_control.html).

Hope this helps,
Stuart

**Reply by cooluser26** [reply]
Hi Stuart, 

Thank you for your quick reply. In 1D, would the timestep simply be `dt = dx / c` ?

**Reply by Status-Mirror** [reply]
Yes, that should be true for the default field-solver.
